pkg/services/ldap/api/service.go

```diff
@@ -327,6 +315,29 @@ func (s *Service) GetUserFromLDAP(c *contextmodel.ReqContext) response.Response
 	return response.JSON(http.StatusOK, u)
 }
 
+func (s *Service) identityFromLDAPUser(user *login.ExternalUserInfo) *authn.Identity {
+	return &authn.Identity{
+		OrgRoles:        user.OrgRoles,
+		Login:           user.Login,
+		Name:            user.Name,
+		Email:           user.Email,
+		IsGrafanaAdmin:  user.IsGrafanaAdmin,
+		AuthenticatedBy: user.AuthModule,
+		AuthID:          user.AuthId,
+		Groups:          user.Groups,
+		ClientParams: authn.ClientParams{
+			SyncUser:            true,
+			SyncTeams:           true,
+			EnableDisabledUsers: true,
+			SyncOrgRoles:        !s.cfg.LDAPSkipOrgRoleSync,
+			AllowSignUp:         s.cfg.LDAPAllowSignup,
+			LookUpParams: login.UserLookupParams{
+				Login: &user.Login,
+			},
```

Comment:
question: looks like we previously used user ID and not login for upserting the user. Why has it changed?

Reply:
good catch, missed that!